CDCgov/prime-reportstream
# Jim/2279 unique blobnames

This PR fixes a bug where if two INTERNAL files (that is, files waiting to be Batched) are created within the same second, we get filename collisions.  The problem is that we are using the external-facing per-state filenames rules/templates, which do not guarantee uniqueness.  LA and OH are examples of naming rules that have failed in Production with this issue.

## Test Steps:

I tested this as follows:

- Create some fake data as follows for LA and OH, and submit to your local docker:

```
./prime data --input-fake 12 --input-schema primedatainput/pdi-covid-19 --output ./junk/foobar.csv --target-states LA,OH,AZ
curl -X POST -H "client:simple_report"  -H "content-type:text/csv" --data-binary "@./junk/foobar.csv" "http://localhost:7071/api/reports"
```

- Then open up your local Azurite and confirm the following:

- Both the la-doh.elr file  and the oh-doh.elr files in the **"reports/batch"** folder should both be of this form:
  `covid-19-a6e0e4a4-7599-4d4b-a178-76cd25ec11c8-20211005164818.internal.csv`
(That file format is the new change in this PR - these .internal.csv filenames now are guaranteed unique)

- The la-doh.elr file in the **"reports/ready"** folder should be of this form:  `cdcprime_cdcprime_laoph_testing_testing_20211005164914.hl7`
The oh-doh.elr file in the "reports/ready" folder should be of this form:  `CDCPRIME_20211005162827.hl7`
(These filenaming rules are requested by the states - so we can't change them)

- And lastly, go into your sftp site and check that the filenames are in this format:

```
sftp -P 2222 foo@localhost
(use pass as your password)
sftp> cd upload
sftp> ls -al

```
- The la-doh and oh-doh files should STILL be in this format style, respectively:
```
cdcprime_cdcprime_laoph_testing_testing_20211005164914.hl7
CDCPRIME_20211005162827.hl7
```


## Changes
- INTERNAL filenames now use a file naming convention that always includes the reportID, so they will always be unique.
- External filenames for download or send are untouched and use state-specific naming conventions, some of which are sadly not well designed, and do not guarantee uniqueness.
- Yes, it is still vaguely possible that files created by the Batch step, following the poor state naming conventions, could have uniqueness problems.   But:  this is exceedingly unlikely because batch only runs occasionally, unlike Receive, which could be getting many submissions per second.

## Checklist

### Testing


## Fixes
- #2279 

# Security

No security changes with this PR.
